pkg/server/registry/apigroups/acorn/events/strategy.go

```diff
@@ -132,15 +161,22 @@ func (q query) filter(events ...apiv1.Event) []apiv1.Event {
 		tail = int(q.tail)
 	}
 
-	if q.prefix.all() {
-		// Query selects all remaining events
-		return events[len(events)-tail:]
-	}
-
 	results := make([]apiv1.Event, 0, tail)
 	for _, event := range events {
+		observed := event.Observed
+		if q.beforeWindow(observed) {
+			// Exclude events observed before the observation window starts
+			continue
+		}
+
+		if q.afterWindow(observed) {
+			// Exclude all events observed after the observation window ends.
+			// Since the slice is sorted chronologically, we can stop filtering here.
+			break
+		}
+
 		if !q.prefix.matches(event) {
```

Comment:
nit: These if statements can be condensed a little bit, at least the ones that just `continue`.

Reply:
done